src/lib/connect/index.ts

```diff
@@ -177,4 +181,64 @@ function connect(
 	return client
 }
 
+function connectAsync(brokerUrl: string): Promise<MqttClient>
```

Comment:
> Or maybe pick a different name for connectAsync to make it clear that it's doing something different from connect

How would you call it?

Reply:
Hmm I'm actually struggling to think of a name. 

I think the problem is that we use `connect()` to instantiate a client, but I think a better API would have the user call the `MqttClient` constructor and then call `connect()` or `connectAsync()` which would call the callback or resolve with a `CONNACK` instead of the client.

Since we're trying to keep the API close to the v4 API, I think we should just keep what's here